File: api/Models/Delivery.cs
using System;

namespace OctoCat.Api.Models
{
    public class Delivery : IEntity<int>
    {
        public int DeliveryId { get; set; }
        public int SupplierId { get; set; }
        public string TrackingNumber { get; set; } = string.Empty;
        public DateTime DeliveryDate { get; set; }
        public string Status { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;

        // Implement IEntity interface
        public int Id 
        { 
            get => DeliveryId;
            set => DeliveryId = value;
        }
    }
}

File: api/src/main/java/com/github/av2/api/model/Delivery.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Delivery information")
public class Delivery {
    @Schema(description = "The unique identifier for the delivery", required = true)
    private Integer deliveryId;
    
    @Schema(description = "The ID of the branch this delivery is for", required = true)
    private Integer branchId;
    
    @Schema(description = "The status of the delivery", required = true)
    private String status;
    
    @Schema(description = "Delivery date and time")
    private String deliveryDate;
    
    @Schema(description = "Additional notes about the delivery")
    private String notes;
}